test/e2e/client_test.go

```diff
@@ -40,21 +44,38 @@ func newTestClient(kubeconfig string) *testClient {
 		panic(err)
 	}
 
+	pc, err := projectclient.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+
 	return &testClient{
 		kc: kc,
+		pc: pc,
 	}
 }
 
-func (t *testClient) createNamespace(namespace string) {
-	// TODO: Create a project request
+func (t *testClient) createNamespace(namespace string) error {
+	if _, err := t.pc.ProjectRequests().Create(&project.ProjectRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		}}); err != nil {
+		return err
+	}
 	t.namespace = namespace
 	// TODO: Wait for a successful SAR check
+	return nil
 }
 
-func (t *testClient) cleanupNamespace(timeout time.Duration) {
+func (t *testClient) cleanupNamespace(timeout time.Duration) error {
 	if t.namespace == "" {
-		return
+		return nil
```

Comment:
nit: simplifying this and have people to defer t.cleanupNamespace() is tempting, but it's not a blocker

Reply:
This is actually part of a AfterEach function so any new tests added should not care about namespace cleanup as it's handled automatically for them.